doc/api/buffer.md

```diff
@@ -203,7 +203,23 @@ The character encodings currently supported by Node.js include:
 
 * `'binary'`: Alias for `'latin1'`.
 
-* `'hex'`: Encode each byte as two hexadecimal characters.
+* `'hex'`: Encode each byte as two hexadecimal characters. Data truncation
+  may occur for unsanitized input. For example:
+
+```js
+Buffer.from('1ag', 'hex');
+// Prints <Buffer 1a>, data truncated when first non-hexa-decimal value
+// ('g') encountered
+
+Buffer.from('1a7g', 'hex');
+// Prints <Buffer 1a> , data truncated when data ends in single digit ('7').
+
+Buffer.from('1634', 'hex');
+// Prints <Buffer 16 34> , fully qualified hexadecimal data
+
+Buffer.from((163).toString(16), 'hex');
+// Prints <Buffer a3> , sanitized hexadecimal data
```

Comment:
I don’t really understand this example – can you explain what “sanitized” means here?

Reply:
@addaleax, In this, I am providing an example where a decimal number is converted to hex through toString(base) API and then passing it to the Buffer API. In this way, input data is guaranteed to be encoded without truncation - as the hex conversion on the user input is performed by toString API already. I termed this as `sanitized` data.

Reply:
@HarshithaKP So… I’ll try to find a good explanation of why I’m not sure that this is a good example:

- For one, this is not a common use case, at least not from what I’ve seen in the wild. I think our documentation should tend to refer to common use cases where possible. And I think for cases in which you do want to write a number into a Buffer, using the `.writeInt...()` or `.writeUInt...()` methods is what I think one would recommend, rather than this approach.
- This type of example only works for integers that don’t end up being truncated because they exceed the `Number.MAX_SAFE_INTEGER` range – for numbers, that’s a pretty large range, but for buffers, that corresponds only a few bytes.
- The term `sanitized` is generally used to refer to input that has been validated and/or stripped of invalid or disallowed data, and loosely associated with security properties of that input; I don’t feel like that’s a good term here.

I’d suggest dropping this example.

Reply:
@addaleax ,
- Agree that it is not a common use case, and hence I put it in the bottom (of common cases)
- The previous examples may cause a normal user to think that only event numbers data is allowed. the fact that this does not apply to numbers in general, but only to those which are passed as hex strings, is not explicit. Or at least that is the understanding I got after reading through original issue. I though this example would help with that
- Agree, `sanitized` is not a good word for this
anyways, I removed this example, following your guidance.